### ISSUE TITLE
version 3.2 not installable on Win7

### ISSUE THREAD

**Issue by mast-eu** [gitextensions/gitextensions#7079]
## Current behaviour

I was testing the MSI for the upcoming 3.2 release. Seems like it can be normally installed on Win10, but **not on Win7**.
Could someone on Win7 please doublecheck?

## Expected behaviour

Installs also on Win7 without issues.

## Steps to reproduce

1. Download the artifact from https://ci.appveyor.com/project/gitextensions/gitextensions/builds/26854222/artifacts. This is build 6443, coresponding to https://github.com/gitextensions/gitextensions/commit/6fe01260fed8720bf9b9a155eb50f53d94b18151, the current tip of branch [release/3.2](https://github.com/gitextensions/gitextensions/tree/release/3.2).
2. Extract the ZIP archive and launch `GitExtensions-3.2.0.6443.msi`, just like you normally would.
3. Go through the configuration dialogs. As far as I can tell, it doesn't matter what options you select. Go with your preferences.
4. After the last dialog, the installer will start installing GitExtension 3.2, but roll back everything before finishing.

## Screenshots

![Untitled](https://user-images.githubusercontent.com/46861028/64052309-51311600-cb7e-11e9-93a7-14aea3c018b1.png)

## Did this work in previous version of GitExtensions

Yes.

## Debug

Launch the installer again from command line with `msiexec.exe /i GitExtensions-3.2.0.6443.msi /L*vx! debug.log`. This creates a detailed log file.

I think the key part is:
```
Action 23:48:35: ConfigureTelemetry. 
MSI (s) (90:08) [23:48:35:762]: Executing op: CustomActionSchedule(Action=ConfigureTelemetry,ActionType=1025,Source=BinaryData,Target=WixQuietExec,CustomActionData="C:\Windows\SysWOW64\WindowsPowerShell\v1.0\powershell.exe" -Version 3.0 -NoProfile -NonInteractive -InputFormat None -ExecutionPolicy Bypass -Command "& 'C:\Program Files (x86)\GitExtensions\set-telemetry.ps1' 1 ; exit $($Error.Count)")
MSI (s) (90:D8) [23:48:35:765]: Invoking remote custom action. DLL: C:\Windows\Installer\MSIC876.tmp, Entrypoint: WixQuietExec
WixQuietExec:  Impossibile avviare Windows PowerShell versione 3.0 perch‚ non Š installato correttamente.
WixQuietExec:  Error 0xfffa0000: Command line returned an error.
WixQuietExec:  Error 0xfffa0000: QuietExec Failed
WixQuietExec:  Error 0xfffa0000: Failed in ExecCommon method
CustomAction ConfigureTelemetry returned actual error code 1603 (note this may not be 100% accurate if translation happened inside sandbox)
Action ended 23:48:35: InstallFinalize. Return value 3.
```
Basically, it is trying to run PowerShell 3.0, which is not available on Win7.

**Comment by RussKie** [comment]
W7 or W7SP1?

**Comment by mast-eu** [comment]
Up-to-date Win7 SP1 (Home Premium) with PowerShell 2.0.

```
> $PSVersionTable

Name                           Value                                           
----                           -----                                           
CLRVersion                     2.0.50727.8806                                  
BuildVersion                   6.1.7601.17514                                  
PSVersion                      2.0                                             
WSManStackVersion              2.0                                             
PSCompatibleVersions           {1.0, 2.0}                                      
SerializationVersion           1.1.0.1                                         
PSRemotingProtocolVersion      2.1  
```

**Comment by RussKie** [comment]
Can you try changing this to PS 2.0 and see if it fixes it?
https://github.com/gitextensions/gitextensions/blob/master/Setup/UI/TelemetryDlg.wxs#L20

**Comment by mast-eu** [comment]
> Can you try changing this to PS 2.0 and see if it fixes it?

I tried, and it did not fix it. However, you helped me to get further: 
Now it's complaining that the method `IsNullOrWhitespace` in https://github.com/gitextensions/gitextensions/blob/149437c93044e49ca20f1277fd9ef609be3cdc8b/Bin/set-telemetry.ps1#L6 hasn't been found. In fact, `IsNullOrWhitespace`  does not exist in PowerShell 2.0. I have `IsNullOrEmpty`, though.
Thus, I changed the line to `[bool]$telemetryEnabled = -not [string]::IsNullOrEmpty($Enabled); `, built the MSI again and ran it on my Win7 machine.

This time, the installation was successful and in `GitExtensions.settings` I find
 ```
 <item>
    <key>
      <string>TelemetryEnabled</string>
    </key>
    <value>
      <string>True</string>
    </value>
  </item>
```

So it's _seems_ to work. I write _seems_, because, using `IsNullOrEmpty` was just a quick shot from the hip:
- I did not test on Win10.
- I did not test to disable telemetry (i.e. to change the passed parameter).
- I am not sure that this will always work as intended.

Maybe tomorrow I can dedicate more time on this, if needed.

**Comment by RussKie** [comment]
> hasn't been found. In fact, `IsNullOrWhitespace` does not exist in PowerShell 2.0. I have `IsNullOrEmpty`, though.
> Thus, I changed the line to `[bool]$telemetryEnabled = -not [string]::IsNullOrEmpty($Enabled); `, built the MSI again and ran it on my Win7 machine.

Great! It is perfectly valid to use `IsNullOrEmpty`, I just made a habbit of using `IsNullOrWhitespace`.

You could also write the line as follows:
```
[bool]$telemetryEnabled = ![string]::IsNullOrEmpty($Enabled);
```

I have done quite a bit of testing of the original change for various use-cases, so it should "just work" (c) 😎

**Comment by RussKie** [comment]
As part of your fix I suggest to add a comment before line 17 in TelemetryDlg.wxs saying something like:

"_\_"

**Comment by mast-eu** [comment]
I could probably send a PR this evening (CET). Do you prefer it to master, release/3.2 or both branches?

**Comment by RussKie** [comment]
3.2 for now, thank you
